zephyr/include/sof/drivers/timer.h

```diff
@@ -14,8 +14,18 @@
 struct comp_dev;
 struct sof_ipc_stream_posn;
 
-#define k_cycle_get_64_safe()		k_cycle_get_64()
-#define k_cycle_get_64_atomic()		k_cycle_get_64()
+#define CYCLES_PER_TICK (sys_clock_hw_cycles_per_sec() / CONFIG_SYS_CLOCK_TICKS_PER_SEC)
+
+static inline uint64_t sof_cycle_get_64(void)
+{
+#if IS_ENABLED(CONFIG_TIMER_HAS_64BIT_CYCLE_COUNTER)
```

Comment:
This use of IS_ENABLED works, but is needless in a preprocessor directive (you might as well use an #ifdef). It's intended for use as a compile time constant in C code, so you can write e.g.:

```c
if (IS_ENABLED(CONFIG_WHATEVER)) {
    ...
} else {
   ...
}
```